lib/avromatic/model/attribute/union.rb

```diff
@@ -0,0 +1,46 @@
+require 'avromatic/model/attribute_type/union'
+
+module Avromatic
+  module Model
+    module Attribute
+      class Union < Virtus::Attribute
+        primitive Avromatic::Model::AttributeType::Union
+
+        def initialize(*)
+          super
+
+          primitive.types.each do |type|
+            member_attributes << Virtus::Attribute.build(type)
+          end
+        end
+
+        def coerce(input)
+          return input if value_coerced?(input)
+
+          result = nil
+          member_attributes.find do |union_attribute|
```

Comment:
Do we want `result = member_attributes.find` to properly handle the case of an invalid attribute (since we assign to `result` before the validation check on line 23)?